### PR TITLE
Adding doc page for DateInput component

### LINKED ISSUE #720: Create doc page for DateInput component

### PR DESCRIPTION
**Pull request by jsuarezgonz**
Closes #720

### CODE REVIEW

File: website/screens/components/date-input/specs/DateInputSpecsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+              <td>Focus indicator</td>
+              <td>
+                <Code>color-blue-600</Code>
+              </td>
+              <td>#0095ff</td>
+            </tr>
+          </tbody>
+        </DxcTable>
+      </DxcStack>
+      <DxcStack gutter="large">
```
Comment (raquelarrojo) [inline website/screens/components/date-input/specs/DateInputSpecsPage.tsx:284]: It is not necessary to add a stack for each table. A stack for the whole design tokens section is enough.

File: website/screens/components/date-input/specs/DateInputSpecsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+        <HeadingLink level={3}>Accessibility</HeadingLink>
+        <HeadingLink level={4}>WCAG 2.2</HeadingLink>
+        <DxcList>
+          <DxcText>
+            Understanding WCAG 2.2 -
+            <DxcLink
+              href="https://www.w3.org/WAI/WCAG22/Understanding/keyboard"
+              text="SC 2.1.1: Keyboard"
+            />
+          </DxcText>
```
Comment (raquelarrojo) [inline website/screens/components/date-input/specs/DateInputSpecsPage.tsx:475]: Add newWindow to the links in accessibility section.

File: website/screens/components/date-input/usage/DateInputUsagePage.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import {
+  DxcText,
+  DxcList,
+  DxcLink,
+  DxcStack,
```
[the remark this answers is not in the document]
Comment (jsuarezgonz) [reply website/screens/components/date-input/usage/DateInputUsagePage.tsx]: Done

File: website/pages/components/date-input/specifications.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import type { ReactElement } from "react";
+import DateInputSpecsPage from "../../../screens/components/date-input/specs/DateInputSpecsPage";
+import DateInputPageLayout from "../../../screens/components/date-input/DateInputPageLayout";
+
+const Specifications = () => {
+  return (
+    <>
+      <Head>
+        <title>Date Input — Halstack Design System</title>
+      </Head>
```
[the remark this answers is not in the document]
Comment (jsuarezgonz) [reply website/pages/components/date-input/specifications.tsx:11]: Done

File: website/screens/components/date-input/specs/DateInputSpecsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+          <Image src={dateSpecs} alt="date input design specifications" />
+        </Figure>
+        <DxcText as="p">
+          The date input <Code>color</Code>, <Code>typography</Code>,{" "}
+          <Code>border</Code>, <Code>spacing</Code>, <Code>width</Code> and{" "}
+          <Code>margin</Code> specifications are inherited from the text input,
+          for reference{" "}
+          <DxcLink
+            href="https://developer.dxc.com/design/guidelines/components/text-input"
+            text="check the text input component documentation"
```
[the remark this answers is not in the document]
Comment (jsuarezgonz) [reply website/screens/components/date-input/specs/DateInputSpecsPage.tsx:35]: Done

File: website/screens/components/date-input/specs/DateInputSpecsPage.tsx (hunk 1 above)
Comment (jsuarezgonz) [reply website/screens/components/date-input/specs/DateInputSpecsPage.tsx]: Done

File: website/screens/components/date-input/specs/DateInputSpecsPage.tsx (hunk 2 above)
Comment (jsuarezgonz) [reply website/screens/components/date-input/specs/DateInputSpecsPage.tsx:439]: Done

**Review by raquelarrojo — APPROVED** [review]

**Pull request merged**